support/cas-server-support-x509-core/src/main/java/org/apereo/cas/adaptors/x509/authentication/principal/X509SubjectAlternativeNameUPNPrincipalResolver.java

```diff
@@ -38,17 +39,31 @@
 @Slf4j
 @ToString(callSuper = true)
 @NoArgsConstructor
+@Setter
 public class X509SubjectAlternativeNameUPNPrincipalResolver extends AbstractX509PrincipalResolver {
 
     /**
      * ObjectID for upn altName for windows smart card logon.
      */
     public static final String UPN_OBJECTID = "1.3.6.1.4.1.311.20.2.3";
 
+    /**
+     * Fallback attribute name that is highly likely to be in the cert (in case processing cert without UPN).
```

Comment:
It would be much better if a smaller variant of your explanation was moved into the javadoc for the actual setting. You could use the scenario here as an example in your explanation of the settings field, if it's especially left in X509Properties. If it's moved to a block of settings that only affect this and that resolver, then copy verbaitm.